xstream/src/java/com/thoughtworks/xstream/io/xml/PrettyPrintWriter.java

```diff
@@ -206,9 +206,7 @@ protected void writeText(final QuickWriter writer, final String text) {
     }
 
     private void writeText(final String text, final boolean isAttribute) {
-        final int length = text.length();
-        for (int i = 0; i < length; i++) {
-            final char c = text.charAt(i);
+        text.codePoints().forEach(c -> {
```

Comment:
I have no idea what you are talking about in this review comment. The method is present in Java 8.

Reply:
Perhaps. Was just going by https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#codePoints() which ways 9. At any rate I would hope the CI build would fail if this were not permitted.

Reply:
> Perhaps

Do you have any evidence for this claim which is casting doubt on the correctness of this change and potentially making it harder for subsequent reviewers to approve? If you do not, I would suggest that you refrain from making such review comments.

https://docs.oracle.com/javase/8/docs/api/java/lang/CharSequence.html#codePoints--

Reply:
The above link. It seems the `@since` tags are contradictory, unless the JDK team has a policy of noting when an override of a default method was added (which would seem strange to me since that should not change the API surface).

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/CharSequence.html#codePoints-- is present in Java 8 and this code compiles successfully on Java 8. As far as I can tell there is no action item here, and this whole review comment was unnecessary and served only to chew up some of my time to refute an unverified claim as well as potentially confusing future reviewers.

Reply:
XStream 1.5.x will target Java 11. No point any longer to use Java 8 as minimum.

Reply:
But XStream 1.4 still uses Java 8, and we want this critical bug fix in that line. Anyway, this change works in Java 8, so this whole thread is pointless. I have no idea why this review feedback was left in the first place.

Reply:
`codePoints()` was added to `CharSequence` interface as a default method in Java 8.
In Java 9, an override of this method was added to `String` (which implements `CharSequence`).

So, it should work for both Java 8 and 9, but it can be slightly faster for Strings in Java 9+ due to optimised version added to `String` in Java 9.